docs/src/upgrade/release-4.x.x.asciidoc

```diff
@@ -30,6 +30,15 @@ complete list of all the modifications that are part of this release.
 
 === Upgrading for Users
 
+==== BulkSet Behavior Changes
+Starting with 4.0, steps which return BulkSet (e.g. `aggregate()`) will have results returned in different format
+depending on embedded or remote usage.
+
+For embedded cases, a BulkSet will be returned as before.
+
+For remote cases, the behavior will be new. Starting with GraphBinaryV4, BulkSet is no longer an IO format, and BulkSet results
```

Comment:
> For remote cases, the behavior will be new. Starting with GraphBinaryV4, BulkSet is no longer an IO format, and BulkSet results returned from the server will be deserialized into Lists in the expanded format.

Can be rephrased to:

For remote cases, the behavior will be new. Starting with GraphBinaryV4, a BulkSet is deserialized and expanded into a List. Each element will in the BulkSet will appear in the list the same number of times as specified by its bulk value.